File: Data Structures & Algorithms/eating-bananas/submission-6.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
         int vremeJela = 0;

        Arrays.sort(piles);
       
        
        int levo = 1;
        int desno = piles[piles.length-1];
        int prethodnoK = 0;
        while (levo<=desno) {
            int vremeZP = 0;

            int medijana = (levo+desno)/2;

            for (int i =0;i<piles.length;i++) {
                int vremeH = piles[i] / medijana;
                int ostatak = piles[i] % medijana;
                if (ostatak != 0)
                    vremeH += 1;
                vremeZP+=vremeH;
            }

            vremeJela=vremeZP;
            if (vremeJela<=h ) {
                desno = medijana-1;
                prethodnoK = medijana;
            }
            else if (vremeJela>h)
                levo = medijana+1;
        }
        return prethodnoK;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-6.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        levo = 0
        desno = len(nums)-1

        if levo==desno and nums[0]==target:
            return 0
        elif levo==desno:
            return -1
        polovina = -1
        while levo<=desno:
            medijana = (levo+desno)//2

            if nums[medijana] > nums[desno]:
                levo = medijana + 1
            elif nums[medijana]<nums[medijana-1]:
                polovina= medijana
                break
            else:
                desno=medijana-1


        levo=0
        desno = polovina-1
        for i in range(0,2):
            while levo<=desno:
                medijana = (levo+desno)//2

                if nums[medijana]<target:
                    levo = medijana+1
                elif nums[medijana]>target:
                    desno = medijana-1
                else:
                    return medijana
            levo = polovina
            desno = len(nums)-1
        return -1